script-library/docker-debian.sh

```diff
@@ -109,6 +110,49 @@ find_version_from_git_tags() {
     echo "${variable_name}=${!variable_name}"
 }
 
+
+# Source /etc/os-release to get OS info
+. /etc/os-release
+# Fetch host/container arch.
+architecture="$(dpkg --print-architecture)"
+
+# Set up the necessary apt repos (either Microsoft's or Docker's)
+if [ "${USE_MOBY}" = "true" ]; then
+
+    cli_package_name="moby-cli"
+
+    # Import key safely and import Microsoft apt repo
+    get_common_setting MICROSOFT_GPG_KEYS_URI
+    curl -sSL ${MICROSOFT_GPG_KEYS_URI} | gpg --dearmor > /usr/share/keyrings/microsoft-archive-keyring.gpg
+    echo "deb [arch=${architecture} signed-by=/usr/share/keyrings/microsoft-archive-keyring.gpg] https://packages.microsoft.com/repos/microsoft-${ID}-${VERSION_CODENAME}-prod ${VERSION_CODENAME} main" > /etc/apt/sources.list.d/microsoft.list
+else
+    # Name of proprietary engine package
+    cli_package_name="docker-ce-cli"
+
+    # Import key safely and import Docker apt repo
+    curl -fsSL https://download.docker.com/linux/${ID}/gpg | gpg --dearmor > /usr/share/keyrings/docker-archive-keyring.gpg
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/${ID} ${VERSION_CODENAME} stable" > /etc/apt/sources.list.d/docker.list
+fi
+
+# Refresh apt lists
+apt-get update
+
+# Soft version matching for CLI
+if [ "${CLI_VERSION}" = "latest" ] || [ "${CLI_VERSION}" = "lts" ] || [ "${CLI_VERSION}" = "stable" ]; then
+    # Empty, meaning grab whatever "latest" is in apt repo
+    cli_version_suffix=""
+else    
+    # Fetch a valid version from the apt-cache (eg: the Microsoft repo appends +azure, breakfix, etc...)
+    cli_version_suffix="=$(apt-cache madison ${cli_package_name} | awk -F"|" '{print $2}' | grep -m 1 "${CLI_VERSION}" | xargs)"
```

Comment:
On second look on the grep statement.  I think to get partial matches to work you'll need to tweak the grep to  use `-E` start at the beginning of the line with "^" and escape ".". You'd then want to add an end of the match being something like "(\\.|$)".  Otherwise you could match a value like "1" anywhere in the string.

Reply:
Same would apply to the docker-in-docker script

Reply:
![image](https://user-images.githubusercontent.com/23246594/135356777-69ef9da9-5f61-40c7-83d5-98694d1e5949.png)

I still think this can be improved (some edge cases when someone supplies something like `2.` - but I think good enough for now!

Reply:
I suspect if you remove the `.+` it will work.  You only need to match on part of the string since you're not passing `-o` - the entire line will get returned.  I can try